signac/contrib/migration/__init__.py

```diff
@@ -31,11 +33,13 @@ def _reload_project_config(project):
 
 def _update_project_config(project, **kwargs):
     """Update the project configuration."""
+    found = False
     for fn in ("signac.rc", ".signacrc"):
-        config = get_config(project.fn(fn))
-        if "project" in config:
-            break
-    else:
+        project_fn = project.fn(fn)
+        if os.path.isfile(project_fn):
+            config = get_config(project_fn)
+            found = True
```

Comment:
Yes, this is the definition that is now present. I am open to alternate suggestions. What sort of use cases would be broken by this definition? I agree that it is weaker than the current definition, but is that necessarily a problem. This is closer to how git (our usual 0th order inspiration) behaves. What configuration information might it be important to include for different subdirectories of a project? If we're thinking about e.g. different FlowProjects, we have previously discussed namespacing flow more precisely around FlowProjects, for example.

Reply:
This makes me very uneasy and I think we need to have a discussion about this.

Reply:
Sure, I'll ping you offline.

Reply:
Closing the loop: @csadorf and I discussed this offline and concluded that the switch planned in #197 will address the problem. Addressing that will move the definition from the existence of a config file to the existence of a `.signac` directory, which is a bit stronger because it avoids edge cases such as treating any directory for centralized config file storage as a signac project. That change may or may not occur in this PR, but it will happen on the `schema2` branch before we merge it into `next`.

Reply:
By "centralized config file storage", do you mean something like `$HOME/.signacrc`? I don't think we've discussed centralized configuration like that in #197 yet, which feels like a loose end we should clarify in that issue. I would guess that we don't want to use a `$HOME/.signac` directory for centralized configuration because it would make the home directory into a signac project, from my understanding of the new definition proposed in #197. I'm happy with pretty much any solution here but would like to hear your thoughts, @vyasr @csadorf.

Reply:
@bdice That's exactly the point. I don't think we plan on abolishing centralized configuration, so with the definition introduced here, the home directory (or any other suitable configuration directly like `~/.config/io.signac/` or `~/Library/Application Support/io.signac/`) would be a signac project which just feels wrong to me. So instead, following the git-approach, any directory that contains a `.signac` sub-directory would be a signac project root directory.